### ISSUE TITLE
Read only MySQL (MongoDB BI connector) results in backtrace

### ISSUE THREAD

**Issue by jonathanunderwood** [airbnb/superset#2751]
Make sure these boxes are checked before submitting your issue - thank you!

- [x] I have checked the superset logs for python stacktraces and included it here as text if any
- [x] I have reproduced the issue with at least the latest released version of superset
- [x] I have checked the issue tracker for the same issue and I haven't found one similar

###
I am trying to use SuperSet to connect to an instance of the MongoDB BI connector which presents a read-only MySQL server, supporting only the commands given in[1]. I believe this should work, but unfortunately the database fails to be added with a backtrace given below.

[1] https://docs.mongodb.com/bi-connector/master/supported-operations/

### Superset version
0.17.6. Full output of pip freeze for the virtualenv below. This is on RHEL7 using the python 2.7 software collection. Same end result if using the python 3.5 software collection.

```
alembic==0.9.1
amqp==1.4.9
anyjson==0.3.3
appdirs==1.4.3
Babel==2.4.0
backports.shutil-get-terminal-size==1.0.0
billiard==3.3.0.23
boto3==1.4.4
botocore==1.5.47
celery==3.1.23
cffi==1.10.0
click==6.7
colorama==0.3.9
cryptography==1.7.2
decorator==4.0.11
docutils==0.13.1
enum34==1.1.6
Flask==0.12.1
Flask-AppBuilder==1.8.1
Flask-Babel==0.11.2
Flask-Cache==0.13.1
Flask-Login==0.2.11
Flask-Migrate==2.0.3
Flask-OpenID==1.2.5
Flask-Script==2.0.5
Flask-SQLAlchemy==2.0
Flask-Testing==0.6.2
Flask-WTF==0.14.2
future==0.16.0
futures==3.1.1
gunicorn==19.7.1
humanize==0.5.1
idna==2.5
ipaddress==1.0.18
ipython==5.3.0
ipython-genutils==0.2.0
itsdangerous==0.24
Jinja2==2.9.6
jmespath==0.9.2
kombu==3.0.37
Mako==1.0.6
Markdown==2.6.8
MarkupSafe==1.0
mysqlclient==1.3.10
numpy==1.12.1
packaging==16.8
pandas==0.19.2
parsedatetime==2.0
pathlib2==2.2.1
pexpect==4.2.1
pickleshare==0.7.4
prompt-toolkit==1.0.14
ptyprocess==0.5.1
pyasn1==0.2.3
pycparser==2.17
pydruid==0.3.1
Pygments==2.2.0
PyHive==0.3.0
pyparsing==2.2.0
python-dateutil==2.6.0
python-editor==1.0.3
python-openid==2.2.5
pytz==2017.2
requests==2.13.0
s3transfer==0.1.10
sasl==0.2.1
scandir==1.5
simplegeneric==0.8.1
simplejson==3.10.0
six==1.10.0
SQLAlchemy==1.1.9
SQLAlchemy-Utils==0.32.14
sqlparse==0.2.3
superset==0.17.6
thrift==0.10.0
thrift-sasl==0.2.1
traitlets==4.3.2
wcwidth==0.1.7
Werkzeug==0.12.1
WTForms==2.1
```

### Expected results
Database is added as new (read only) data source

### Actual results
The following traceback occurs:
```python
/opt/superset27/lib/python2.7/site-packages/sqlalchemy/engine/default.py:305: SAWarning: Exception attempting to detect unicode returns: ProgrammingError('(_mysql_exceptions.ProgrammingError) (1064, "parse sql \'SELECT CAST(\'test plain returns\' AS CHAR(60)) AS anon_1\' error: syntax error at position 42")',)
  "detect unicode returns: %r" % de)
/opt/superset27/lib/python2.7/site-packages/sqlalchemy/engine/default.py:305: SAWarning: Exception attempting to detect unicode returns: ProgrammingError('(_mysql_exceptions.ProgrammingError) (1064, "parse sql \'SELECT CAST(\'test unicode returns\' AS CHAR(60)) AS anon_1\' error: syntax error at position 44")',)
  "detect unicode returns: %r" % de)
/opt/superset27/lib/python2.7/site-packages/sqlalchemy/engine/default.py:305: SAWarning: Exception attempting to detect unicode returns: ProgrammingError('(_mysql_exceptions.ProgrammingError) (1064, "parse sql \'SELECT CAST(\'test collated returns\' AS CHAR CHARACTER SET utf8) COLLATE utf8_bin AS anon_1\' error: syntax error at position 55 near character")',)
  "detect unicode returns: %r" % de)
2017-05-11 15:56:43,490:ERROR:root:(_mysql_exceptions.OperationalError) (1105, 'statement *parser.Rollback not supported')
Traceback (most recent call last):
  File "/opt/superset27/lib/python2.7/site-packages/superset/views/core.py", line 1343, in testconn
    engine.connect()
  File "/opt/superset27/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 2091, in connect
    return self._connection_cls(self, **kwargs)
  File "/opt/superset27/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 90, in __init__
    if connection is not None else engine.raw_connection()
  File "/opt/superset27/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 2177, in raw_connection
    self.pool.unique_connection, _connection)
  File "/opt/superset27/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 2151, in _wrap_pool_connect
    e, dialect, self)
  File "/opt/superset27/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 1465, in _handle_dbapi_exception_noconnection
    exc_info
  File "/opt/superset27/lib/python2.7/site-packages/sqlalchemy/util/compat.py", line 203, in raise_from_cause
    reraise(type(exception), exception, tb=exc_tb, cause=cause)
  File "/opt/superset27/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 2147, in _wrap_pool_connect
    return fn()
  File "/opt/superset27/lib/python2.7/site-packages/sqlalchemy/pool.py", line 328, in unique_connection
    return _ConnectionFairy._checkout(self)
  File "/opt/superset27/lib/python2.7/site-packages/sqlalchemy/pool.py", line 766, in _checkout
    fairy = _ConnectionRecord.checkout(pool)
  File "/opt/superset27/lib/python2.7/site-packages/sqlalchemy/pool.py", line 516, in checkout
    rec = pool._do_get()
  File "/opt/superset27/lib/python2.7/site-packages/sqlalchemy/pool.py", line 1138, in _do_get
    self._dec_overflow()
  File "/opt/superset27/lib/python2.7/site-packages/sqlalchemy/util/langhelpers.py", line 66, in __exit__
    compat.reraise(exc_type, exc_value, exc_tb)
  File "/opt/superset27/lib/python2.7/site-packages/sqlalchemy/pool.py", line 1135, in _do_get
    return self._create_connection()
  File "/opt/superset27/lib/python2.7/site-packages/sqlalchemy/pool.py", line 333, in _create_connection
    return _ConnectionRecord(self)
  File "/opt/superset27/lib/python2.7/site-packages/sqlalchemy/pool.py", line 461, in __init__
    self.__connect(first_connect_check=True)
  File "/opt/superset27/lib/python2.7/site-packages/sqlalchemy/pool.py", line 661, in __connect
    exec_once(self.connection, self)
  File "/opt/superset27/lib/python2.7/site-packages/sqlalchemy/event/attr.py", line 246, in exec_once
    self(*args, **kw)
  File "/opt/superset27/lib/python2.7/site-packages/sqlalchemy/event/attr.py", line 256, in __call__
    fn(*args, **kw)
  File "/opt/superset27/lib/python2.7/site-packages/sqlalchemy/util/langhelpers.py", line 1331, in go
    return once_fn(*arg, **kw)
  File "/opt/superset27/lib/python2.7/site-packages/sqlalchemy/engine/strategies.py", line 181, in first_connect
    dialect.initialize(c)
  File "/opt/superset27/lib/python2.7/site-packages/sqlalchemy/dialects/mysql/base.py", line 1682, in initialize
    default.DefaultDialect.initialize(self, connection)
  File "/opt/superset27/lib/python2.7/site-packages/sqlalchemy/engine/default.py", line 266, in initialize
    self.do_rollback(connection.connection)
  File "/opt/superset27/lib/python2.7/site-packages/sqlalchemy/dialects/mysql/base.py", line 1575, in do_rollback
    dbapi_connection.rollback()
OperationalError: (_mysql_exceptions.OperationalError) (1105, 'statement *parser.Rollback not supported')
```

### Steps to reproduce
1. Install and run superset and the mysqlclient python packages
2. Try adding a new database under Sources, entering a URI of the form mysql://uname:passwd@bicon.domain.com:3307/somedatabase
3. Click test connection
4. Backtrace above results

**Comment by jonathanunderwood** [comment]
Digging in further, this boils down to the BI connector not supporting a sufficient set of SQL-99 commands for the MySQL SQLAlchemy DBAPI layer to work - the lack of TRANSACTION support in particular. I'll close this, as I don't think it can be fixed by SuperSet devs.

**Comment by aguspina** [comment]
Hey @jonathanunderwood ! I'm trying to to connect my superset instance to a mongodb bi connector. I'm having the same issue as you. Do you have any advice for me? Thanks!

**Comment by jonathanunderwood** [comment]
> Hey @jonathanunderwood ! I'm trying to to connect my superset instance to a mongodb bi connector. I'm having the same issue as you. Do you have any advice for me? Thanks!

@aguspina I gave up. I contacted MongoDB support at the time and asked them to add the required TRANSACTION support for this, but that was rejected at the time, as the point of the BI connector is meant to only provide read functionality. That was 3 years ago, and I've moved on to other technologies now for that use case. Sorry to be of no help to you :(
